controllers/ocirepository_controller.go

```diff
@@ -942,3 +980,73 @@ func (r *OCIRepositoryReconciler) notify(ctx context.Context,
 		}
 	}
 }
+
+// Verify verifies the authenticity of the given image reference url. First, it tries to keyful approach by looking at
+// whether the given secret exists. Then, if it does not exist, it pushes a keyless approach for verification.
+func (r *OCIRepositoryReconciler) verify(ctx context.Context, obj *sourcev1.OCIRepository, url string) (bool, error) {
+	// get the public keys from the given secret
+	secretRef := obj.Spec.Verify.SecretRef
+
+	// Generate the registry credential keychain either from static credentials or using cloud OIDC
+	keychain, err := r.keychain(ctx, obj)
+	if err != nil {
+		return false, err
+	}
+
+	authnKeychain := signature.WithAuthnKeychain(keychain)
+
+	ref, err := name.ParseReference(url)
+	if err != nil {
+		return false, err
+	}
+
+	if secretRef != nil {
+		ctrl.LoggerFrom(ctx).Info(fmt.Sprintf("secretRef found: %s", secretRef.Name))
+		certSecretName := types.NamespacedName{
+			Namespace: obj.Namespace,
+			Name:      secretRef.Name,
+		}
+
+		var pubSecret corev1.Secret
+		if err := r.Get(ctx, certSecretName, &pubSecret); err != nil {
+			return false, err
+		}
+
+		// traverse all public keys and try to verify the signature
+		// this is brute-force approach, but it is ok for now
+		for k, data := range pubSecret.Data {
+			// search for public keys in the secret
+			if strings.HasSuffix(k, ".pub") {
+				verifier, err := signature.New(signature.WithPublicKey(data), authnKeychain)
+				if err != nil {
+					return false, err
+				}
+
+				signatures, _, err := verifier.VerifyImageSignatures(ctx, ref)
+				if err != nil {
+					return false, err
+				}
+
+				if len(signatures) > 0 {
+					return true, nil
+				}
+			}
+		}
+	} else {
+		ctrl.LoggerFrom(ctx).Info(fmt.Sprintf("secretRef not found: %s, trying keyless verification.", secretRef.Name))
```

Comment:
This will panic, there is no secret to log its name. I don’t think this function should emit info logs, we don’t want to spam uses at every reconciliation if there is nothing wrong with the verification process.

Reply:
Thx for the warning, I missed that part, fixed now.